kyo-data/shared/src/main/scala/kyo/Result.scala

```diff
@@ -148,86 +110,129 @@ object Result:
       *
       * @param either
       *   The Either to convert
-      * @tparam E
-      *   The type of the error (Left value)
-      * @tparam A
-      *   The type of the successful value (Right value)
       * @return
       *   A Result equivalent to the input Either
       */
     def fromEither[E, A](either: Either[E, A]): Result[E, A] =
-        either.fold(fail, success)
+        either.fold(fail, succeed)
 
     /** Converts a Try to a Result.
       *
       * @param t
       *   The Try to convert
-      * @tparam A
-      *   The type of the successful value
       * @return
       *   A Result equivalent to the input Try
       */
-    def fromTry[A](t: Try[A]): Result[Throwable, A] =
-        t.fold(fail, success)
+    def fromTry[A](t: Try[A]): Result[Throwable, A] = t.fold(fail, succeed)
+
+    /** Collects a sequence of Results into a single Result containing a sequence of successful values.
+      *
+      * @param seq
+      *   The sequence of Results to collect
+      * @return
+      *   A Result containing either a sequence of successful values or the first encountered error
+      */
+    def collect[E, A](seq: Seq[Result[E, A]]): Result[E, Seq[A]] =
+        def loop(remaining: Seq[Result[E, A]], acc: Chunk[A]): Result[E, Chunk[A]] =
+            remaining match
+                case (head: Result[E, A]) +: tail =>
+                    head.flatMap(value => loop(tail, acc.append(value)))
+                case Seq() => Success(acc)
+
+        try loop(seq, Chunk.empty[A])
+        catch
+            case ex => Panic(ex)
+    end collect
+
+    /** Evaluates an expression, catching a specific exception type.
+      *
+      * @param expr
+      *   The expression to evaluate
+      * @return
+      *   A Result containing either the successful value, a Failure with the caught exception, or a Panic for other exceptions
+      */
+    inline def catching[E <: Throwable](
+        using inline ct: SafeClassTag[E]
+    )[A](inline expr: => A): Result[E, A] =
+        try
+            Success(expr)
+        catch
+            case ct(ex)             => Failure(ex)
+            case ex if NonFatal(ex) => Panic(ex)
```

Comment:
I'm starting to get used to clause interleaving. Maybe we could migrate the codebase to it?